archive/app.py

```diff
@@ -0,0 +1,153 @@
+import boto3
+import datetime
+import logging
+import os
+from django.db import transaction  # type: ignore
+from openstates.cli.reports import generate_session_report
+import urllib.parse
+
+logger = logging.getLogger("openstates")
+s3_client = boto3.client("s3")
+s3_resource = boto3.resource("s3")
+
+
+def archive_processed_file(bucket, key):
+    """
+    Archive the processed file to avoid possible scenarios of race conditions.
+    We currently use meta.client.copy instead of client.copy b/c it can copy
+    multiple files via multiple threads, since we have batching in view.
+
+    Args:
+        bucket (str): The s3 bucket name
+        key (str): The key of the file to be archived
+    Returns:
+        None
+
+    Example:
+        >>> archive_processed_file("my-bucket", "my-file.json")
+    """
+
+    global s3_resource
```

Comment:
Same comment John and I made on app.py is applicable here on archive/app.py about globals

Reply:
Will clear it out.